src/Elastic.Apm/Model/Payload/Service.cs

```diff
@@ -24,7 +25,7 @@ internal static Service GetDefaultService(IConfigurationReader configurationRead
 				Agent = new AgentC
 				{
 					Name = Consts.AgentName,
-					Version = Consts.AgentVersion
+					Version = Assembly.GetExecutingAssembly().GetName().Version?.ToString() ?? "n/a",
```

Comment:
Might be better to use:
```
typeof(AgentC).Assembly.GetName().Version?.ToString()
```

Reply:
👍  ...ended up using the `Agent` type, since these model classes can easily end up in a shared library (with different versioning) and the `Agent` would be probably always in the core lib.